runtimes/peregrine/src/asset_switch/mod.rs

```diff
@@ -42,19 +41,30 @@ impl SwitchHooks<Runtime, KiltToEKiltSwitchPallet> for RestrictswitchDestination
 		_amount: Balance,
 	) -> Result<(), Self::Error> {
 		let to_as_v3: MultiLocation = to.clone().try_into().map_err(|e| {
-			log::error!(target: LOG_TARGET, "Failed to convert beneficiary Multilocation {:?} to v3 with error {:?}", to, e);
+			log::error!(
+				target: LOG_TARGET,
+				"Failed to convert beneficiary Multilocation {:?} to v3 with error {:?}",
+				to,
+				e
+			);
 			Error::Internal
 		})?;
-		ensure!(
-			to_as_v3.interior
-				== Junction::AccountId32 {
-					network: None,
-					id: from.clone().into()
-				}
-				.into(),
-			Error::NotToSelf
-		);
-		Ok(())
+		let is_beneficiary_self = to_as_v3.interior
+			== Junction::AccountId32 {
+				network: None,
+				id: from.clone().into(),
+			}
+			.into();
+		cfg_if::cfg_if! {
+			if #[cfg(feature = "runtime-benchmarks")] {
+				// Clippy complaints the variable is not used with this feature on, otherwise.
+				let _ = is_beneficiary_self;
+				Ok(())
+			} else {
+				frame_support::ensure!(is_beneficiary_self, Error::NotToSelf);
+				Ok(())
+			}
+		}
```

Comment:
Why is this needed? Can we not specify, in the benchmarks, that the receiver is the sender?

Reply:
You mean using the same account in the benchmarks to avoid doing this here?

Reply:
yes

Reply:
I changed the logic, and allow the implementor to pass more info, none of them required. https://github.com/KILTprotocol/kilt-node/pull/671/commits/b9d28e5376ad30f1aae95f9b3992ad45572bc298

I like it much more now, thanks very much for the suggestions.

Reply:
Maybe it is overengineered, but what do you think of using the [derive_builder](https://docs.rs/derive_builder/latest/derive_builder/)?

Reply:
Mhh yeah but in the benchmarks we still want to convert any `None` into an actual value, so we still need a struct that has `Option`s, and one that does not. The builder pattern would be applied to the `OptionalBenchmarkInfo` struct, but we still internally need to convert that into a `BenchmarkInfo` one. Not sure it would actually bring that much value, given that this is info the trait implementer should return. If they need or wish, they could have a builder on one of their type, and then simply implement `Into<OptionalBenchmarkInfo>` for `MyType`.

Reply:
I personally think the trait should be as agnostic as possible over the details of its implementer. So if they want to use a builder pattern, they're free to do so. All the trait should care, is the return type of the implementation.

Reply:
I see okay